AberrantSMPP/ASyncSocketClient.cs

```diff
@@ -329,9 +348,12 @@ public void Send(byte[] buffer)
 #endif
 
 						//send the data; don't worry about receiving any state information back;
-						_SendPending = true; 
-						_TcpClient.GetStream()
-							.BeginWrite(buffer, 0, buffer.Length, _CallbackWriteMethod, null);
+						_SendPending = true;
+
+						if (_UseSsl)
+							_SslStream.BeginWrite(buffer, 0, buffer.Length, _CallbackWriteMethod, null);
```

Comment:
I think it is going to be better to hold the Stream (SslStream or TcpClient's native Stream) at a private field (ie _stream), and use it all along the code of this class. Otherwise you are going to be repeating this same pattern of (if useSsl do this else do that).

Reply:
Okey, as you mentioned, I use now a private field to hold the stream (_Stream) instead of repeating the same pattern of (if useSsl do this else do that).